### PR TITLE
Allow travel pay claims with null appointment date values, remove cla…

### PR DESCRIPTION
**Pull request by athifw**
…im sorting

## Summary

Travel pay claims with `null` `appointmentDateTime` were causing errors when sorting in the `get_claims` method. This PR allows such claims to pass through. Sorting claims is already being done in the frontend, so sorting claims here is redundant.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/89328

## Testing done

- [x] *New code is covered by unit tests*
- Travel pay claims shown in frontend (https://github.com/department-of-veterans-affairs/vets-website/pull/31348)

## Screenshots
Screenshot 2024-08-12 at 1 25 56 PM

### CODE REVIEW

**Review by hinzed1127 — DISMISSED** [review]
I'm fine removing the BE sorting for now unless @kjduensing has objections

File: modules/travel_pay/spec/services/service_spec.rb [hunk 1]
```diff
@@ -52,8 +62,8 @@
     end
 
     it 'returns sorted and parsed claims' do
```
Comment (kjduensing) [inline modules/travel_pay/spec/services/service_spec.rb:64]: I'm fine with removing sorting, but if we're doing that, we should:
* remove anything to do with IDs, they were only there to ensure ordering
* expect title-cased statuses in a non-ordered way

File: spec/support/vcr_cassettes/travel_pay/200_claims.yml [hunk 2]
```diff
@@ -179,7 +179,7 @@ http_interactions:
             "modifiedOn": "2024-04-23T16:44:34.465Z"
           },
```
Comment (RachalCassity) [inline spec/support/vcr_cassettes/travel_pay/200_claims.yml:180]: Can you please replace the URI with an variable that can be defined in the [vcr spec](https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/support/vcr.rb#L3).

**Review by RachalCassity — DISMISSED** [review]

File: spec/support/vcr_cassettes/travel_pay/200_claims.yml (hunk 2 above)
Comment (kjduensing) [reply spec/support/vcr_cassettes/travel_pay/200_claims.yml:180]: Can I clarify? You want us to:

* Add this URL to the vcr spec: https://auth.veis.gov/tenant_id/oauth2/token
* Replace all instances of the that url with the variable in this file

Is that right @RachalCassity?

Comment (RachalCassity) [reply spec/support/vcr_cassettes/travel_pay/200_claims.yml:180]: @kjduensing Yes, please. At least replace the domain, https://auth.veis.gov/.

Comment (kjduensing) [reply spec/support/vcr_cassettes/travel_pay/200_claims.yml:180]: Thanks for clarifying, Rachal! 

cc @athifw

**Review by kjduensing — APPROVED** [review]
Looks good!

**Review by ericboehs — APPROVED** [review]

**Review by kjduensing — APPROVED** [review]
Looks good again

**Pull request merged**